### ISSUE TITLE
Bug: Maven complains `JAVA_HOME` environment variable is not defined correctly

### ISSUE THREAD

**Issue by Philzen** [sdkman/sdkman-cli#1234]
**Bug report**
Just installed sdkman as per the instructions and ran `sdk install …` to get latest java 8 & 11 (Temurin) and Maven 3.8.7.

As a matter of precaution, i de-installed all maven and java versions (as far as possible, some programs on my machine depend on the latest version). Even restarted the machine. But no matter what i try, maven (which is now the sdkman version) complains:

```terminal
$ mvn
The JAVA_HOME environment variable is not defined correctly,
this environment variable is needed to run this program.
```

**To reproduce**
🤔

**System info**

- Manjaro Linux with Kernel 6.1.49-1
- GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)
- SDKMAN version: script: 5.18.2, native: 0.4.2
- `sdk current` → java: 11.0.20.1-tem, maven: 3.8.7
- `echo $JAVA_HOME` → /home/myusername/.sdkman/candidates/java/11.0.20.1-tem

**Comment by Philzen** [comment]
BTW – `.sdkmanrc` with `sdkman_auto_env=true` works like a charm (which was the killer feature that makes me consider advising the whole dev team of my customer to switch to it):

```terminal
$ cd project-to-migrate/

Using maven version 3.8.7 in this shell.

Using java version 8.0.382-tem in this shell.
$ echo $JAVA_HOME
/home/myusername/.sdkman/candidates/java/8.0.382-tem
```

**Comment by Philzen** [comment]
Oh bugger: https://stackoverflow.com/a/60838522/1246547 :facepalm: 

# TL;DR: Check your `~/.maven_rc` file
